archetype/engine-v2/src/main/java/io/helidon/build/archetype/engine/v2/InputResolver.java

```diff
@@ -176,4 +182,24 @@ public VisitResult postVisitAny(Input input, Context context) {
         }
         return VisitResult.CONTINUE;
     }
+
+    @Override
+    public VisitResult visitValidation(Validation validation, Context arg) {
+        validationId = validation.id();
```

Comment:
How about storing the current list of regex instead of the current validation id.
This way we can avoid the `map.get` call in the `visitRegex` method.

```java
this.regexs = new LinkedList<>();
this.validations.put(validation.id(), this.regexps);
```